plugin/kubernetes/setup.go

```diff
@@ -287,11 +275,68 @@ func ParseStanza(c *caddy.Controller) (*Kubernetes, error) {
 				continue
 			}
 			return nil, c.ArgErr()
+		case "external":
+			args := c.RemainingArgs()
+			if len(args) > 0 {
+				k8s.opts.expose = exposeAll
+				for _, a := range args {
+					extZone := plugin.Host(a).Normalize()
+					found := false
+					for _, z := range k8s.Zones {
+						if dnsutil.IsReverse(z) > 0 {
+							continue
+						}
+						if z == extZone {
+							found = true
+							break
+						}
+					}
+					if !found {
+						return nil, c.Errf("external zone '%s' is not in plugin/block zone list", c.Val())
```

Comment:
none of the plugins have these checks, I'm also fan of garbage in, garbage out, so we should add it here.
As the corefile has all context we can easily write a tools that checks the config, ala named-checkconfig instead of pulling out that code directly into coredns